src/bases/createWidgetBase.ts

```diff
@@ -68,22 +69,22 @@ function realizeDNode(instance: Widget<WidgetState>, dNode: DNode) {
 			instance.own(child);
 		}
 		if (!id && internalState.currentChildrenMap.has(factory)) {
-			console.error('must provide unique keys when using the same widget factory multiple times');
+			const errorMsg = 'must provide unique keys when using the same widget factory multiple times';
+			console.error(errorMsg);
```

Comment:
Think about this again, it would be good to create an `Error` here, but not throw it, so downstream they get the stack trace etc and return that in the emit as well as what gets logged.  I think we also need to think about how this fits into `dojo-debug` (cc: @rishson) and if we actually want to log errors to the console or if we will go with some other sort of instrumentation that can be easier to manage.